### PR TITLE
mat64: extend TestInverse

### LINKED ISSUE #412: mat64: Dense.Inverse test coverage needs improvement
There is no testing for matrices that are not RawMatrixer or matrices that are a transposed RawMatrixer.

### PR DESCRIPTION
**Pull request by kortschak**
@vladimir-ch @btracey Please take a look for discussion.

The single failing case is one where I have ignored the shadowing restrictions, so we do not guarantee that it will work. I am not entirely sure how I want to handle this. I think that probably leaving the test case in as a comment explaining why it would fail is the correct answer. What are your views?

Fixes #412.

### CODE REVIEW

**Comment by ChristopherRabotin** [comment]
I don't see any test on singular matrices, which is how #410 got to slip through the cracks. Maybe a few should be added?

**Comment by kortschak** [comment]
No, I'll leave that for @vladimir-ch in #411. This set of tests is to ensure that our typish system works properly.

File: mat64/dense_test.go [hunk 1]
```diff
@@ -1633,6 +1634,56 @@ func TestInverse(t *testing.T) {
...
+				0.147222222222222, -0.144444444444444, 0.063888888888889,
+				-0.061111111111111, 0.022222222222222, 0.105555555555556,
+				-0.019444444444444, 0.188888888888889, -0.102777777777778,
+			}),
+			tol: 1e-14,
+		},
+
+		// The following test will fail since it does not obey the shadowing
+		// rules. The basicMatrix type allows shadowing of the input data
+		// without providing the Raw mathod required for detection of shadowing.
```
Comment (btracey) [inline mat64/dense_test.go]: This is a bit of a contrived example, which is fine, but I think the comment could be adjusted to make that more clear. 

"The following case fails as it disobeys the shadowing rules. Specifically, the test below extracts the underlying *Dense, and uses it as a recevier with the basicMatrix as input. The basicMatrix type allows shadowing of the input data without providing the Raw mathod required for detection of shadowing. We specifically state we cannot check this case."

File: mat64/dense_arithmetic.go [hunk 2]
```diff
@@ -228,15 +228,7 @@ func (m *Dense) Inverse(a Matrix) error {
...
-		if m != aU {
-			m.Copy(a)
-		} else if aTrans {
-			// m and a share data so Copy cannot be used directly.
-			tmp := getWorkspace(r, c, false)
-			tmp.Copy(a)
-			m.Copy(tmp)
-			putWorkspace(tmp)
-		}
+		m.Copy(a)
```
Comment (btracey) [inline mat64/dense_arithmetic.go]: Does this work for, say RawTriangular? It doesn't look like it would to me

**Comment by btracey** [comment]
I think leaving the test commented is good. It's probably nice to have a few shadowing examples somewhere when we have somewhere to put that kind of extra documentation.

File: mat64/dense_arithmetic.go (hunk 2 above)
Comment (kortschak) [inline mat64/dense_arithmetic.go]: What do you mean does it work for RawTriangular? The rules say that we won't check for shadowing unless "the Raw type matches that of the receiver", which it doesn't here.

Comment (btracey) [inline mat64/dense_arithmetic.go]: Good point. Nice cleanup then.

**Comment by kortschak** [comment]
PTAL

**Comment by btracey** [comment]
LGTM

Sorry for the delay, I missed you had fixed.

**Pull request merged**